docs/CODE_GUIDELINES.md

```diff
@@ -341,28 +358,49 @@ int const * const someIntPointer;
 
 ### 5.4. Initialization
 
-Make sure that variables are initialized appropriately at declaration or soon afterwards. This is especially important for fundamental type variables that do not have any constructor. Zero-initialize with `{}`.
+Make sure that variables are initialized appropriately at declaration or soon afterwards. This is especially important for fundamental type variables that do not have any constructor to zero-initialize with `{}`.
 
 ✅ Good:
 ```cpp
-int x{};
-int* y{};
-CLog::Log("test: {} {}", x, y);
+int x{3};
+CLog::Log("test: {}", x); // variable used just after its declaration
 ```
 
 ❌ Bad:
 ```cpp
-int x; // used uninitialized
-int* y = nullptr; // default-initialization not using {}
-CLog::Log("test: {} {}", x, y);
+int x{3}; // variable not immediately used, if possible declare the variable close to the point of its use 
+[...many lines of code that do not use variable x...]
+CLog::Log("test: {}", x);
 ```
 
-In general, prefer the `{}` initializer syntax over alternatives. This syntax is less ambiguous and disallows narrowing (cf. [ISO C++ guidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-list)).
+Although we suggest adopting the use of `{}` for the local variables, like the example above,
+for this use case we allow to initialize the variables also by using the form `=` to improve the code readability.
 
 ✅ Good:
 ```cpp
-int x{5};
-int y{x};
+std::string localVariableX{m_fooVariableOne + m_barVariableTwo}; // less readable
+std::string localVariableX = m_fooVariableOne + m_barVariableTwo; // more readable
```

Comment:
I think what is more readable is mostly a matter of habit, but IMO the guidelines should pick one as a recommendation (can also be the second option for all I care) and not describe both as "good" just to immediately judge them then ("less readable"). Why should anyone want to use the worse option then? :D  if there is a preference, write it down. if not, don't judge.

Reply:
as my reply above, if you prefeer you can suggest to add a  better explanation for this difference,
or open a discussion on Slack and get everyone involved to discuss this difference another time

Reply:
i can remove the example if you mean this

Reply:
i have reworked all this section, i think will fit better your requests

Reply:
alrighty, thanks!